x-pack/plugins/global_search_providers/public/providers/get_app_results.ts

```diff
@@ -117,6 +148,8 @@ const flattenDeepLinks = (
         ...deepAppLink,
         // shift current sublink title into array of sub-sublink titles
         subLinkTitles: [deepLink.title, ...deepAppLink.subLinkTitles],
+        // combine current sublink keywords into array of sub-link keywords
+        subLinkKeywords: deepLink.meta.keywords.concat(deepAppLink.subLinkKeywords),
```

Comment:
We need to keep a record of all the keywords used for the apps and those that are registered via deep links.